gravitee-am-common/src/main/java/io/gravitee/am/common/utils/ConstantKeys.java

```diff
@@ -104,6 +104,9 @@ public interface ConstantKeys {
     String MFA_ENROLLMENT_EMAIL = "email";
     long DEFAULT_ENROLLMENT_SKIP_TIME_SECONDS = 10 * 60 * 60; // ten hours
 
+    String MFA_STOP = "mfaStop";
```

Comment:
**thought**: we already have 2 other keys to deal with MFA state (MFA_ENROLLMENT_COMPLETED_KEY &  MFA_CHALLENGE_COMPLETED_KEY ). Should we consider a single key "MFA_FLOW_STATE" with an enum of state (enrolling, enrolled, challenged, stop...)... wdyt ? 
If we decide to do it, it may have an impact on some plugin like the [mfa-challenge](https://github.com/gravitee-io/gravitee-am-policy-mfa-challenge/blob/master/src/main/java/com/graviteesource/am/policy/mfa/challenge/MfaChallengePolicy.java#L122) one... and we will have to continue to manage the MFA_ENROLLMENT_COMPLETED_KEY &  MFA_CHALLENGE_COMPLETED_KEY in the session to not invalidate the active session of enduser when the 4.3.0 will be deployed...

Reply:
what would be the purpose of it?

Reply:
I'm thinking of the number of attributes in the context. We are clearly managing states to manage the MFA FLow. Having a single "state" attribute may simplify the readability on long term purpose. Don't say we have to do it, just a thought. The attributes we are setting in the session are various and I think we should start to rethink the way the session info are used. I was thinking this MFA "state" was a good candidate but maybe not and you may think the session is good enough :)

Reply:
ok, we can think about it and change it in the next tasks. In this, there are already a lot of changes.